### ISSUE TITLE
Rails/ScopeArgs classifies `lambda` as a plain method call

### ISSUE THREAD

**Issue by backus** [bbatsov/rubocop#2954]
## Expected behavior

This scope should not violate the `Rails/ScopeArgs` cop:

```ruby
class SomeModel < ActiveRecord::Base
  scope :my_scope, lambda do |param|
    where(blah: param)
  end
end
```

## Actual behavior

```sh
$ bundle exec rubocop --rails --only Rails/ScopeArgs app/models/example.rb
Inspecting 1 file
C

Offenses:

app/models/example.rb:2:20: C: Use lambda/proc instead of a plain method call.
  scope :my_scope, lambda do |param|
                   ^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

Here is a gist you can clone: https://gist.github.com/backus/644bbfe9c411e227b63a

Just make sure that you move the `example.rb` to `app/models/example.rb`. After that you should be able to just run `./example.sh`

## RuboCop version

```
$ bundle exec rubocop -V
0.38.0 (using Parser 2.3.0.6, running on ruby 2.3.0 x86_64-darwin15)
```

**Edit**: I opened a PR fixing the issue: https://github.com/bbatsov/rubocop/pull/2955

**Comment by mikegee** [comment]
Does it work though?  I get: `tried to create Proc object without a block (ArgumentError)` because `do`/`end` style blocks give the block to the higher method call (`scope` here) and `lambda` gets no block.

**Comment by backus** [comment]
> Does it work though? I get: tried to create Proc object without a block (ArgumentError) because do/end style blocks give the block to the higher method call (scope here) and lambda gets no block.

Sigh you might be right actually I can't believe I didn't check that. I'll close this issue for now and reopen if there is still valid syntax which isn't checked.

**Comment by backus** [comment]
I should have spotted this issue before: https://github.com/bbatsov/rubocop/issues/1520
